modules/standard/GMP.chpl

```diff
@@ -1222,9 +1222,9 @@ module GMP {
     }
   }
 
-  /* FUTURE -- GMP numbers with record semantics,
-      expression and operator overloads.
-  */
+  if CHPL_GMP == "none" {
+    compilerError("GMP support not compiled in");
```

Comment:
I'm open to different error messages -- this is just based on a similiar error message for regexp support: https://github.com/chapel-lang/chapel/blob/master/modules/standard/Regexp.chpl#L414

Reply:
This error is better than no error, but I'd probably want something that helped a newbie user help themselves more (in both GMP and Regexp cases).  Referring directly to `CHPL_GMP` being set to `none` might help them find what they need with a docs search.  Or referring to the need to set CHPL_GMP to a value other than none and rebuilding might also be useful.

Reply:
Do you like any of these?

 - "GMP support is not available with CHPL_GMP=none"
 - "Cannot use GMP with CHPL_GMP=none"
 - "Cannot use GMP with GMP_GMP=none, see https://chapel-lang.org/docs/master/usingchapel/chplenv.html#chpl-regexp for more information"


I don't love hardcoding a link, but I also don't love saying something like "set CHPL_GMP and rebuild" because it doesn't feel obvious that CHPL_GMP is an env variable, and it also doesn't seem like it's obvious that we mean rebuild the chpl runtime/third-party vs "recompile" your app

Reply:
Heads up, you're linking against chpl-regexp, not chpl-gmp

Reply:
Maybe the second one?  I could also imagine changing it to say "the GMP module" rather than just "GMP".

It occurs to me belatedly that we should probably add a similar error in the BigInteger module.

Reply:
Wouldn't the BigInteger module benefit from the GMP error, since it uses the GMP module?

Reply:
Or do you mean that the error should be more explicit, since it isn't necessarily clear that BigInteger relies on GMP?

Reply:
Exactly.  And I was thinking / hoping that putting a more explicit error into BigInteger.chpl would cause it to print first (though now I'm worrying that that's incorrect...).

Reply:
Since BigInteger uses GMP you get the compilerError from the gmp module, and I couldn't emit a different error message for BigInteger, even if I put the compilerError before the [`use GMP;`](https://github.com/chapel-lang/chapel/blob/master/modules/standard/BigInteger.chpl#L112).

That's part of the reason I was trying to leave the error a little more vague and not say "the gmp module"

Reply:
I see.  OK, let's go with this then and wait for users to express confusion before trying to do something more involved.